tools/src/main/scala/com/fulcrumgenomics/str/vcf/StrGenotypeDuplexMolecules.scala

```diff
@@ -0,0 +1,357 @@
+    |  3. the name of the STR (ex. D1S1656)
+    |  4. optionally, the expected (known or truth) number of repeat units for allele #1
+    |  5. optionally, the expected (known or truth) number of repeat units for allele #2
+    |
+    |The output will contain a single sample with a diploid genotype call.  The INFO field will have list the allele
+    |frequencies for all alleles observed, not just those in the final genotype.
+    |
+    |A PDF with plots will be output, containing a plot per STR showing the counts observed repeat unit length.  If the
+    |expected number of repeat units are given, a horizontal dotted line will be plotted for each expected allele.
+  """)
+class StrGenotypeDuplexMolecules
+(
+  @arg(flag='i', doc="Input VCF of somatic variant calls.") val input: PathToVcf,
+  @arg(flag='o', doc="Prefix for all output files.") val output: PathPrefix,
+  @arg(flag='l', doc="Interval list with the STR regions.") val intervals: PathToIntervals
+) extends FgStrTool with LazyLogging {
+  import StrGenotypeDuplexMolecules.StrInfo
+
+  // TODO: do we want to output no-calls for missing sites?
+
+  private val ScriptPath = "com/fulcrumgenomics/str/vcf/StrGenotypeDuplexMolecules.R"
+
+  Io.assertReadable(input)
+  Io.assertReadable(intervals)
+  Io.assertCanWriteFile(output)
+
+  override def execute(): Unit = {
+
+    def f(ext: String): FilePath = PathUtil.pathTo(output + ext)
+
+    val vcfIn           = new VCFFileReader(input.toFile, false)
+    val overlapDetector = loadIntervals()
+    val sampleName      = toSampleName(vcfIn.getFileHeader)
+    val vcfOut          = toVcfWriter(header=vcfIn.getFileHeader, sampleName=sampleName, output=f(".vcf.gz"))
+    val infoWriter      = Io.toWriter(f(".info.txt"))
+    val progress        = ProgressLogger(logger=logger, noun="variants")
+
+    vcfIn.foreach { ctx =>
+      val overlaps = overlapDetector.getOverlaps(ctx).toSeq
+      if (overlaps.isEmpty) {
+        logger.warning(s"Skipping variant call with no overlaps: $ctx")
+      }
+      else {
+        // Developer notes:
+        // - at a some point in the future, it would be nice to remove all of the INFO and FORMAT values we no longer
+        //   use
+
+        val strInfo = overlaps match {
+          case Seq()     => unreachable("Found an emtpy overlap")
+          case Seq(info) => info
+          case infos     =>
+            logger.warning(s"Found multiple overlaps for variant: $ctx")
+            infos.head
+        }
+
+        val ctxBuilder = new VariantContextBuilder(ctx)
+        val genotypes  = ctx.getGenotypes
+        val alleles    = ctx.getAlleles.toSeq
+
+        // get the count per allele
+        val counts: Seq[Int] =  alleles.map { allele => genotypes.map { genotype => genotype.countAllele(allele) }.sum }
+        require(alleles.length == counts.length, s"Found ${alleles.length} alleles and ${counts.length} counts")
```

Comment:
This seems impossible to fail.  Seq.map is guaranteed to produce a Seq of the same length.